### PR TITLE
More accessibility - better Narrator support, keyboard navigation

### PR DESCRIPTION
**Pull request by livcheerful**
fixes https://github.com/microsoft/pxt-microbit/issues/3629
fixes  https://github.com/microsoft/pxt-microbit/issues/3605
fixes https://github.com/microsoft/pxt-microbit/issues/3603
fixes https://github.com/microsoft/pxt-microbit/issues/3596

should have made a separate commit for each one D:

### CODE REVIEW

File: webapp/src/share.tsx [hunk 1]
```diff
@@ -515,7 +515,8 @@ export class ShareEditor extends data.Component<ShareEditorProps, ShareEditorSta
...
                         {!qrCodeFull && <sui.Input id="projectUri" class="mini" readOnly={true} lines={1} value={url} copy={true} autoFocus={!pxt.BrowserUtils.isMobile()} selectOnClick={true} aria-describedby="projectUriLabel" autoComplete={false} />}
                         {!qrCodeFull && <label htmlFor="projectUri" id="projectUriLabel" className="accessible-hidden">{lf("This is the read-only internet address of your project.")}</label>}
-                        {!!qrCodeUri && <img className={`ui ${qrCodeFull ? "huge" : "small"} image ${qrCodeExpanded ? "centered" : "floated right"} button pixelart`} alt={lf("QR Code of the saved program")} src={qrCodeUri} onClick={this.handleQrCodeClick} title={lf("Click to expand or collapse.")} />}
+                        {!!qrCodeUri && <img className={`ui ${qrCodeFull ? "huge" : "small"} image ${qrCodeExpanded ? "centered" : "floated right"} button pixelart`} alt={lf("QR Code of the saved program")}
+                            src={qrCodeUri} onClick={this.handleQrCodeClick} title={lf("Click to expand or collapse.")} tabIndex={0} aria-label={lf("QR Code of the saved program")}/>}
```
Comment (jwunderl) [inline webapp/src/share.tsx:519]: does this need a sui.fireClickOnEnter to be able to expand/collapse with keyboard or does it work as is?

**Review by jwunderl — APPROVED** [review]
LGTM, I don't see a fix for https://github.com/microsoft/pxt-microbit/issues/3596 but I'm pretty sure I fixed that in https://github.com/microsoft/pxt/pull/8062/files#diff-3e7cafbd95631be2faaeb5d32fe7bfc6b5d33f707af4d40d2067ac479d7953c9R411 so maybe merging master got rid of it?

**Comment by livcheerful** [comment]
oh yeah master already had the button role so I removed my change

File: webapp/src/share.tsx (hunk 1 above)
Comment (livcheerful) [reply webapp/src/share.tsx:519]: you right you right

**Review by shakao — APPROVED** [review]

**Pull request merged**